### PR TITLE
doc(go): improve SDK quick start

### LINKED ISSUE #117: Demo `PackOptions` explicitly
### Code of Conduct

- [X] I have read and agree to the ORAS' [Code of Conduct](https://github.com/oras-project/oras-www/blob/main/CODE_OF_CONDUCT.md)

### What part of document/web-page on oras.land is affected?

In the page https://oras.land/client_libraries/0_go/, `oras.PackOptions` is demostrated along with a large comment block.

We can merge the comment block into the code for a simpler doc.

### Additional comments?

_No response_

### PR DESCRIPTION
**Pull request by shizhMSFT**
Resolves #117

### CODE REVIEW

File: docs/Client_Libraries/0_go.md [hunk 1]
```diff
@@ -120,11 +122,11 @@ func pushFiles() {
 ### Pull files from a remote repository
 
 ```go
-func pullFiles() {
+func pullFiles() error {
```
Comment (Wwwsylvia) [inline docs/Client_Libraries/0_go.md:125]: Just realized that the `pullFiles` example is incomplete (missing imports, etc.).  Let me add it after this PR is merged.

**Review by Wwwsylvia — APPROVED** [review]
LGTM

File: docs/Client_Libraries/0_go.md (hunk 1 above)
Comment (shizhMSFT) [reply docs/Client_Libraries/0_go.md:125]: I can add them in this PR.

Comment (shizhMSFT) [reply docs/Client_Libraries/0_go.md:138]: QED

**Review by TerryHowe — APPROVED** [review]
LGTM

**Review by FeynmanZhou — APPROVED** [review]
LGTM

**Pull request merged**